Blog for HA (#62)
aiven/klaw-docs: docs/blog/2023/07/high-availability-for-klaw.rst

High availability for Klaw
==========================

.. post:: July 4, 2023
   :tags: kafka,governance,topics,acls,klaw, high availability, fault tolerance
   :category: klaw
   :author: Murali Basani

Background
----------

> In today's dynamic and ever-changing digital landscape, maintaining the uninterrupted operation of applications is of the highest importance. This article explores how to configure Klaw for high availability in production environments, enhancing reliability and delivering a seamless user experience.
> 
> In this blog post, we will dive deep into Klaw—an Java-based application explicitly created for Kafka governance. We will focus on exploring the configuration process for achieving high availability (HA) in production environments. By implementing these measures, we can ensure continuous operation, bolster reliability, and deliver a seamless user experience, even in unexpected failures.

Reviewer: ```suggestion
In today's dynamic and ever-changing digital landscape, maintaining the uninterrupted operation of applications is of the highest importance. This blog post explores the configuration of Klaw, a Java-based application designed specifically for Kafka governance. Our primary focus is on exploring the configuration process for achieving high availability (HA) in production environments. By implementing these measures, we can ensure continuous operation, enhance reliability, and deliver a seamless user experience, even during unexpected failures.
```

> In today's dynamic and ever-changing digital landscape, maintaining the uninterrupted operation of applications is of the highest importance. This article explores how to configure Klaw for high availability in production environments, enhancing reliability and delivering a seamless user experience.
> 
> In this blog post, we will dive deep into Klaw—an Java-based application explicitly created for Kafka governance. We will focus on exploring the configuration process for achieving high availability (HA) in production environments. By implementing these measures, we can ensure continuous operation, bolster reliability, and deliver a seamless user experience, even in unexpected failures.

Reviewer: Shortened this section to improve clarity. 
```suggestion
In today's dynamic and ever-changing digital landscape, maintaining the uninterrupted operation of applications is of the highest importance. This blog post explores the configuration of Klaw, a Java-based application designed specifically for Kafka governance. 

Our primary focus is on exploring the configuration process for achieving high availability (HA) in production environments. By implementing these measures, we can ensure continuous operation, enhance reliability, and deliver a seamless user experience, even during unexpected failures.
```

Reviewer: @muralibasani update the content for the "Background" section. We used a mix of words like article and blogs, just wanted to keep the words consistent.

> In this blog post, we will dive deep into Klaw—an Java-based application explicitly created for Kafka governance. We will focus on exploring the configuration process for achieving high availability (HA) in production environments. By implementing these measures, we can ensure continuous operation, bolster reliability, and deliver a seamless user experience, even in unexpected failures.

Reviewer 2: Sorry missed this the first time
"Klaw—an Java-based" -> Klaw—a Java-based

High availability for Klaw
==========================

> 

Reviewer 2: Just noticed this is repeated twice one in the Mya folder and one in july.

Below is an example Nginx configuration using Round-Robin load balancing:

**Nginx configuration**

::

    upstream klawgcp {
        round-robin; // load balancing method
        server klawhost:9097;
        server klawhost:9097;
    }
    server {
            listen 80;
            listen [::]:80;
            ssl on;

            ssl_certificate     /opt/klaw/certs/service.cert;
            ssl_certificate_key /opt/klaw/certs/service.key;
            ssl_protocols TLSv1.2;
            server_name serverhost;
            location / {
                    proxy_pass https://klawgcp;
                }
    }

Klaw deployment model in high availability (HA) mode
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In the high availability mode, Nginx routes the requests to Klaw instances which are connected to a common data store such as Postgres. Each Klaw instance comprises both the Governance and Cluster API applications. You may choose to deploy these applications on separate machines for increased robustness.
Below is the deployment model of Klaw in HA mode.

> Klaw in HA mode with Nginx Load balancer

Reviewer: @muralibasani Just realized that there is an image missing here. Apart from this, the rest looks good.

Author: @harshini-rangaswamy thank you. Added the image now.
